pkg/conduit/pipelines_parser.go

```diff
@@ -0,0 +1,111 @@
+// Copyright © 2022 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package conduit
+
+import (
+	"io"
+
+	"github.com/conduitio/conduit/pkg/foundation/cerrors"
+	"gopkg.in/yaml.v3"
+)
+
+const ParserVersion = "1.0"
+
+type ProcessorConfig struct {
+	Settings map[string]string `yaml:"settings"`
+}
+
+type ProcessorInfo map[string]struct {
+	Name   string          `yaml:"name"`
+	Type   string          `yaml:"type"`
+	Config ProcessorConfig `yaml:"config"`
+}
+
+type ConnectorConfig struct {
+	Name     string            `yaml:"name"`
+	Settings map[string]string `yaml:"settings"`
+}
+
+type ConnectorInfo map[string]struct {
+	Type       string          `yaml:"type"`
+	Plugin     string          `yaml:"plugin"`
+	Config     ConnectorConfig `yaml:"config"`
+	Processors ProcessorInfo   `yaml:"processors,omitempty"`
+}
+
+type PipelineConfig struct {
+	Name        string `yaml:"name"`
+	Description string `yaml:"description"`
+}
+
+type PipelineInfo struct {
+	Status     string         `yaml:"status"`
+	Config     PipelineConfig `yaml:"config"`
+	Connectors ConnectorInfo  `yaml:"connectors,omitempty"`
+	Processors ProcessorInfo  `yaml:"processors,omitempty"`
+}
+
+type PipelinesInfo struct {
+	Version   string                  `yaml:"version"`
+	Pipelines map[string]PipelineInfo `yaml:"pipelines"`
+}
+
+func Parse(data io.Reader) (PipelinesInfo, error) {
+	dec := yaml.NewDecoder(data)
+
+	var docs []PipelinesInfo
+	for {
+		var doc PipelinesInfo
+		err := dec.Decode(&doc)
```

Comment:
How does the decoder handle unknown fields? Does it ignore them or does it fail?

Reply:
it ignores them